python/hail/table.py

```diff
@@ -1248,7 +1256,12 @@ def joiner(obj):
         return construct_expr(GlobalJoinReference(uid), self.global_schema,
                               joins=LinkedList(Join).push(Join(joiner, [uid])))
 
-    @typecheck_method(exprs=Expression)
+    def _check_field_name(self, name, indices):
```

Comment:
I'd prefer we put this in a separate file and pass in `self._fields` rather than duplicate it.

Reply:
will do